README

```diff
@@ -53,11 +53,33 @@ You can also use the image standalone (without TFTP under Foreman's
 control). In this case, edit your pxelinux.cfg/default file directly and
 make sure the foreman.url points correctly.
 
+Extending the image at runtime with a ZIPfile
+---------------------------------------------
+
+It's possible to provide a zip containing extra code for the image to use.
+First, create a directory structure like:
+
+```
+./bin
+./facts
+./lib
+./lib/ruby
+```
+
+`bin` is added to PATH, you can place binaries here. If this directory
+contains an executable `autostart` script, it will be run *before* the
+image is registered to Foreman. `facts` is added to FACTERLIB so that
+custom facts may be configured and sent to Foreman. `lib` is added to
+LD_LIBRARY_PATH and `lib/ruby` is added to RUBYLIB, so that binaries in
+`bin` can be executed properly.
+
+Currently, the location of this is hardcoded - the image will look for
+`#{foreman.url}/extension.zip` so place the zip in your Foreman public/
+directory (TODO, add proper routes to the proxy for this).
```

Comment:
I don't like this hardcoding. Let's just do the ext=HTTP it's straightforward. Users can choose to use proxy/foreman or maybe a different HTTP server on the network.